File: allforone-java/src/main/java/com/example/allforone_java/MainActivity.java
package com.example.allforone_java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import bean.newss;
import config.HttpSubscribe;
import netUtils.OnSuccessAndFaultListener;
import netUtils.OnSuccessAndFaultSub;

public class MainActivity extends AppCompatActivity {

    private Context mContext;
    private TextView mTvGet;
    private TextView mTvPost;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mContext = this;
        initViews();
    }

    private void initViews() {
        mTvGet = (TextView) findViewById(R.id.tv_get);
        mTvPost = (TextView)findViewById(R.id.tv_post);

        mTvGet.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                HttpSubscribe.getNews(mContext,"top","9f552a8aca577737335c7106f1236a97",new OnSuccessAndFaultSub<newss>(new OnSuccessAndFaultListener<newss>() {
                    @Override
                    public void onSuccess(newss result) {
                        mTvGet.setText(result.getResult().getData().get(0).getTitle());
                    }

                    @Override
                    public void onFault(String errorMsg) {
                        Log.e("zbf","错误信息:"+errorMsg);
                    }
                }));
            }
        });

        mTvPost.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                HttpSubscribe.postNews(mContext,"top","9f552a8aca577737335c7106f1236a97",new OnSuccessAndFaultSub<newss>(new OnSuccessAndFaultListener<newss>() {
                    @Override
                    public void onSuccess(newss result) {
                        mTvPost.setText(result.getResult().getData().get(0).getTitle());
                    }

                    @Override
                    public void onFault(String errorMsg) {
                        Log.e("zbf","错误信息:"+errorMsg);
                    }
                }));
            }
        });
    }
}

File: app/src/main/java/com/allforone/MainActivity.kt
package com.allforone

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.allforone.bean.newss
import com.allforone.http.NetSubscribe
import com.allforone.http.listener.OnSuccessAndFaultListener
import com.allforone.http.listener.OnSuccessAndFaultSub
import com.allforone.ktx.activity
import com.allforone.ktx.logE
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Author : zofnk.
 * Email : zofnk@vip.qq.com.
 * Creat Time :  11.21. 20:12
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()
    }

    private fun initViews() {

        tv_get.setOnClickListener {
            NetSubscribe.getNews(
                activity,
                "top",
                "9f552a8aca577737335c7106f1236a97",
                OnSuccessAndFaultSub(object : OnSuccessAndFaultListener<newss> {

                    override fun onSuccess(result: newss) {
                        tv_get.text = result.result.data[0].title
                    }

                    override fun onFault(errorMsg: String) {
                        Log.e("zbf", "错误信息:$errorMsg")
                        "错误信息:$errorMsg".logE()
                    }
                })
            )
        }

        tv_post.setOnClickListener {
            NetSubscribe.postNews(
                activity,
                "top",
                "9f552a8aca577737335c7106f1236a97",
                OnSuccessAndFaultSub(object : OnSuccessAndFaultListener<newss> {
                    override fun onSuccess(result: newss) {
                        tv_post.text = result.result.data[0].title
                    }

                    override fun onFault(errorMsg: String) {
                        "错误信息:$errorMsg".logE()
                    }
                })
            )
        }
    }
}